tests/PHPStan/Type/TypeCombinatorTest.php

```diff
@@ -2009,6 +2010,46 @@ public function dataUnion(): iterable
 				UnionType::class,
 				'literal-string|lowercase-string',
 			],
+			[
+				[
+					new StringType(),
+					new IntersectionType([new StringType(), new AccessoryUppercaseStringType()]),
+				],
+				StringType::class,
+				'string',
+			],
+			[
+				[
+					new IntersectionType([new StringType(), new AccessoryNumericStringType()]),
+					new IntersectionType([new StringType(), new AccessoryUppercaseStringType()]),
+				],
+				UnionType::class,
+				'numeric-string|uppercase-string',
+			],
+			[
+				[
+					new IntersectionType([new StringType(), new AccessoryNonFalsyStringType()]),
+					new IntersectionType([new StringType(), new AccessoryUppercaseStringType()]),
+				],
+				UnionType::class,
+				'non-falsy-string|uppercase-string',
+			],
+			[
+				[
+					new IntersectionType([new StringType(), new AccessoryNonEmptyStringType()]),
+					new IntersectionType([new StringType(), new AccessoryUppercaseStringType()]),
+				],
+				UnionType::class,
+				'non-empty-string|uppercase-string',
+			],
+			[
+				[
+					new IntersectionType([new StringType(), new AccessoryLiteralStringType()]),
+					new IntersectionType([new StringType(), new AccessoryUppercaseStringType()]),
+				],
+				UnionType::class,
+				'literal-string|uppercase-string',
```

Comment:
You should add a test for lowercase-string|uppercase-string
I assumed this should be simplified to string.

Reply:
Added in 96773a4d2